searx/engines/ask.py

```diff
@@ -0,0 +1,56 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+"""Ask.com"""
+
+from urllib.parse import urlencode
+import re
+from lxml import html
+
+# Metadata
+about = {
+    "website": "https://www.ask.com/",
+    "wikidata_id": 'Q847564',
+    "official_api_documentation": None,
+    "use_official_api": False,
+    "require_api_key": False,
+    "results": "HTML",
+}
+
+# Engine Configuration
+categories = ['general']
+paging = True
+
+# Base URL
+base_url = "https://www.ask.com/web"
+
+
+def request(query, params):
+
+    query_params = {
+        "q": query,
+        "page": params["pageno"],
+    }
+
+    params["url"] = f"{base_url}?{urlencode(query_params)}"
+    return params
+
+
+def response(resp):
+
+    text = html.fromstring(resp.text).text_content()
+    urls_match = re.findall(r'"url":"(.*?)"', text)
+    titles_match = re.findall(r'"title":"(.*?)"', text)[3:]
+    content_match = re.findall(r'"abstract":"(.*?)"', text)
```

Comment:
grep a regexp in a XML (HTML) content is fragile (consider the regexp pattern is in the result itself / or in the search term) .. like in all the other engines we should use xpath selectors.

A good starting point are the xpath selectors. I have stolen from @allendema's PR https://github.com/searx/searx/pull/2898 / A XPATH engine that we have never merged into SearXNG (was merged into the old project searx but not here in this project SearXNG)


```python

    title_xpath = '//a[@class="PartialSearchResults-item-title-link result-link"]'
    url_xpath = '//a[@class="PartialSearchResults-item-title-link result-link"]/@href'
    content_xpath = '//p[@class="PartialSearchResults-item-abstract"]'
    suggestion_xpath = '//span[@class="PartialRelatedSearch-item-link-text"]'
```